### PR TITLE
Add an initial draft of lolopy

### PR DESCRIPTION
**Pull request by WardLT**
Shows training a model and getting uncertainties out.

### CODE REVIEW

File: python/lolopy/learners.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+        # Get the model out
+        self.model_ = result.getModel()
+
+        return self
+
+    @abstractmethod
+    def _make_learner(self):
+        """Instantiate the learner used by Lolo to train a model
+
+        Ret"""
```
Comment (maxhutch) [inline python/lolopy/learners.py]: Is this incomplete?

File: python/lolopy/learners.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+            # Copy the X into an array
+            x_i_java = ListConverter().convert(x_i, self.gateway._gateway_client)
+            x_i_java = self.gateway.jvm.scala.collection.JavaConverters.asScalaBuffer(
+                x_i_java).toVector()
+
+            # If a regression problem, make sure y_i is a float
+            if is_regressor(self):
+                y_i = float(y_i)
+            else:
+                y_i = int(y_i)
```
Comment (maxhutch) [inline python/lolopy/learners.py]: Does this imply that string labels are not supported?

File: python/lolopy/learners.py (hunk 1 above)
Comment (WardLT) [reply python/lolopy/learners.py]: It sure is. Fixing it now

File: python/lolopy/learners.py [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+            # If weights are None, RF expects a 3ple
+            if weights is None:
+                pair = self.gateway.jvm.scala.Tuple3(x_i_java, y_i, 1.0)
```
Comment (maxhutch) [inline python/lolopy/learners.py]: Why not incorporate the weights here?

File: python/lolopy/learners.py [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+    def fit(self, X, y, weights=None):
+        # Get the number of classes
+        self.n_classes_ = len(set(y))
+
+        return super(RandomForestClassifier, self).fit(X, y, weights)
+
+    def predict(self, X):
+        return super(RandomForestClassifier, self).predict(X, False)
+
+    def predict_proba(self, X):
```
Comment (maxhutch) [inline python/lolopy/learners.py]: `predict_proba`?  Did you mean `predict_probability`?

File: python/setup.py [hunk 5]
```diff
@@ -0,0 +1,8 @@
+from setuptools import setup
+
+setup(
+    name='lolopy',
+    version='0.0.1',
+    packages=['lolopy'],
+    description='Python wrapper for the Lolo machine learning library',
```
Comment (maxhutch) [inline python/setup.py]: We should add sklearn and py4j to install_requirements.

File: python/lolopy/learners.py (hunk 4 above)
Comment (WardLT) [reply python/lolopy/learners.py]: No, `predict_proba` is the proper function name for scikit-learn classifiers.

File: python/lolopy/learners.py (hunk 2 above)
Comment (WardLT) [reply python/lolopy/learners.py]: That is correct. I need the class labels to have a defined ordering, and the easiest route to that is to use integers and employ the [LabelEncoder](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html) built in to ScikitLearn for strings, etc.

File: python/setup.py (hunk 5 above)
Comment (WardLT) [reply python/setup.py]: Done
